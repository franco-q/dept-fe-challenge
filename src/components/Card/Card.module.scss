@import '@/scss/styles.scss';

.root {
  background: linear-gradient(
      0deg,
      rgba(255, 255, 255, 0.05),
      rgba(255, 255, 255, 0.05)
    ),
    #121212;
  box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.25);
  border-radius: 8px;
  display: flex;
  font-family: 'D-DIN';
  font-style: normal;
  font-weight: 400;
  margin-bottom: 16px;
  text-decoration: none;
  &,
  &:active,
  &:hover {
    color: #ffffffde;
  }
}

.figure {
  height: 109px;
  width: 121px;
  display: block;
  margin: 0;
  overflow: hidden;
  img {
    display: block;
    object-fit: contain;
    height: 100%;
    width: 100%;
  }
}

.content {
  width: calc(100% - 121px);
  padding-left: 12px;
  padding-top: 11px;
  padding-right: 9px;
  padding-bottom: 10px;
}

.title {
  font-size: 16px;
  line-height: 24px;
  margin-bottom: 4px;
}

.description {
  font-size: 12px;
  line-height: 18px;
  height: 37px;
  overflow: hidden;
  margin: 0;
}

.date {
  font-size: 12px;
  line-height: 18px;
  letter-spacing: 0.03em;
  color: rgba(255, 255, 255, 0.37);
}

.footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.favBtn {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 24px;
  width: 24px;
  padding: 0;
  background-color: #00000000;
  border: none;
}

@include media-breakpoint-up(tablet) {
  .root {
    flex-direction: column;
  }
  .figure {
    height: 158px;
    width: 100%;
    border-radius: 8px;
  }
  .content {
    width: 100%;
  }
  .title {
    font-size: 20px;
    line-height: 32px;
    font-weight: 700;
  }

  .description {
    font-size: 16px;
    line-height: 24px;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
  }

  .date {
    font-size: 14px;
    line-height: 22px;
  }
}
