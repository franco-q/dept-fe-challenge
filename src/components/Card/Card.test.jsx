import { render, fireEvent } from '@testing-library/react'
import Card from './Card'

jest.mock('react-router-dom', () => ({
  Link: ({ children, to }) => {
    return <a href={to}>{children}</a>
  }
}))

describe('Card', () => {
  test('should render', () => {
    const props = {
      id: 1143239401,
      srcImg: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
      altImg: 'FalconSat',
      title: 'FalconSat',
      description: 'Engine failure at 33 seconds and loss of vehicle',
      date: 'March 24, 2006',
      flightNumber: 1,
      favourite: false
    }

    const { asFragment } = render(<Card {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should execute cb when fav btn is clicked', () => {
    const onClickFav = jest.fn()
    const props = {
      id: 1143239401,
      srcImg: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
      altImg: 'FalconSat',
      title: 'FalconSat',
      description: 'Engine failure at 33 seconds and loss of vehicle',
      date: 'March 24, 2006',
      flightNumber: 1,
      favourite: false
    }
    const { container } = render(<Card {...props} onClickFav={onClickFav} />)

    fireEvent.click(container.querySelector('button'))

    expect(onClickFav).toHaveBeenCalledWith(props.id)
  })
})
