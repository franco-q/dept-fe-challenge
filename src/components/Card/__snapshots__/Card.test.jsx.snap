// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<DocumentFragment>
  <a
    href="/launch/1"
  >
    <figure>
      <img
        alt="FalconSat"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
    </figure>
    <div>
      <h4>
        FalconSat
      </h4>
      <p>
        Engine failure at 33 seconds and loss of vehicle
      </p>
      <div>
        <span>
          March 24, 2006
        </span>
        <button>
          <svg
            fill="none"
            height="11"
            viewBox="0 0 13 11"
            width="13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M6.26325 8.9678C6.41108 8.88833 6.58892 8.88833 6.73675 8.9678L9.50564 10.4563C9.89039 10.6631 10.3368 10.3118 10.2261 9.88923L9.53489 7.25028C9.48403 7.05612 9.55422 6.85036 9.71313 6.72774L11.9681 4.9878C12.3309 4.70784 12.1573 4.1279 11.7003 4.09337L8.6157 3.86027C8.43685 3.84676 8.27895 3.73849 8.20188 3.57654L6.95149 0.948815C6.77057 0.568602 6.22943 0.568602 6.04851 0.948815L4.79812 3.57654C4.72105 3.73849 4.56315 3.84676 4.3843 3.86027L1.29968 4.09337C0.842688 4.1279 0.669071 4.70784 1.03191 4.9878L3.28687 6.72774C3.44578 6.85036 3.51597 7.05612 3.46511 7.25028L2.77393 9.88922C2.66325 10.3118 3.10961 10.6631 3.49436 10.4563L6.26325 8.9678Z"
              fill=""
              fill-opacity="0.87"
              fill-rule="evenodd"
              stroke="white"
            />
          </svg>
        </button>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
